server/http.go

```diff
@@ -868,3 +936,47 @@ func (p *Plugin) slackAttachmentToUpdatePMI(currentValue, channelID string) *mod
 
 	return &slackAttachment
 }
+
+func (p *Plugin) checkChannelPreference(channelID string) (bool, int, error) {
+	channel, appErr := p.API.GetChannel(channelID)
+	if appErr != nil {
+		return false, http.StatusInternalServerError, appErr
+	}
+
+	zoomChannelSettingsMap, err := p.listZoomChannelSettings()
+	if err != nil {
+		return false, http.StatusInternalServerError, err
+	}
+
+	val, exist := zoomChannelSettingsMap[channelID]
+	preference := false
+	/*
+		Check if zoom settings for current channel exist.
+		Check if creating meeting is disabled in the plugin configuration.
+	*/
+	if exist {
+		if val.Preference == ZoomChannelPreferences[DefaultPreference] {
+			preference = p.configuration.RestrictBotPosting
+		} else if val.Preference == ZoomChannelPreferences[EnablePreference] {
+			preference = true
+		}
+	} else {
+		if channel.Type == model.ChannelTypeOpen {
+			preference = p.configuration.RestrictBotPosting
+		}
+	}
+
+	return preference, http.StatusOK, nil
```

Comment:
If the channel type is private and no preference is set for it, we will be returning `false` in that case. Is this expected?

Reply:
yes, by returning `false`, we are not restricting user from creating meeting links in private channel at line https://github.com/mattermost/mattermost-plugin-zoom/blob/6837578decc382b65d1acdb0f1541d6232cde73a/server/http.go#L510